### PR TITLE
mon/PGMap: fix osd_epoch update when removing osd_stat

### PR DESCRIPTION
**Pull request by liewegas**
Fixes: http://tracker.ceph.com/issues/20208 (one instance of it at least)

### CODE REVIEW

**Comment by dachary** [comment]
FWIW, passed ceph-disk suite at https://github.com/ceph/ceph/pull/15576#issuecomment-307221142 (with a minor ceph-disk fix)

**Comment by yuriw** [comment]
http://pulpito.ceph.com/yuriw-2017-06-08_19:40:51-rados-wip-yuri-testing2_2017_7_10-distro-basic-smithi

http://pulpito.ceph.com/yuriw-2017-06-09_00:46:51-rados-wip-yuri-testing2_2017_7_10-distro-basic-smithi/

File: src/mon/PGMap.cc [hunk 1]
```diff
@@ -1130,6 +1130,7 @@ void PGMap::apply_incremental(CephContext *cct, const Incremental& inc)
     if (t != osd_stat.end()) {
       stat_osd_sub(t->first, t->second);
       osd_stat.erase(t);
+      osd_epochs.erase(*p);
```
Comment (Liuchang0812) [inline src/mon/PGMap.cc]: Jenkins is broken at `flush_pg_status` always. is it relevant?

**Comment by dachary** [comment]
https://jenkins.ceph.com/job/ceph-pull-requests/25879/console also: I suspect this is in master actually.

**Comment by Liuchang0812** [comment]
@dachary you are right. `make check` with master is failed.

**Pull request merged**